core/primitives/src/views.rs

```diff
@@ -127,13 +127,13 @@ impl From<AccountView> for Account {
 
 impl From<ContractCode> for ContractCodeView {
     fn from(contract_code: ContractCode) -> Self {
-        ContractCodeView { code: contract_code.code, hash: contract_code.hash }
+        ContractCodeView { code: contract_code.code().to_vec(), hash: *contract_code.hash() }
```

Comment:
Yeah, this `.to_vec()` is unfortunate, let's add `impl ContractCode { pub fn into_code(self) -> Vec<u8> { self.code } }` to avoid it.

Reply:
then here we'll have
```rust
fn from(contract_code: ContractCode) -> Self {
    let hash = *contract_code.hash();
    ContractCodeView { code: contract_code.into_code(), hash }
}
```
and in `core/store/src/lib.rs`
``` rust
pub fn set_code(state_update: &mut TrieUpdate, account_id: AccountId, code: &ContractCode) {
    state_update.set(TrieKey::ContractCode { account_id }, code.clone().into_code());
}
```

which doesn't look good imo.

Reply:
I'd write the first as

```rust
    let hash = *contract_code.hash();
    let code = contract_code.into_code()
    ContractCodeView { code, hash }
```

That's a bit boilerplate-y, but boilerplate is expected in this layer. What we are doing here is converting from our internal, private types to public JSON representation exposed via HTTP API. Such code just always look like "lets convert a struct to exactly the same struct", but it provides great value -- you can change private stuff without affecting the API. That it does look boilerplaty doesn't meant that we should drop perf on the floor ;)

For the second case, you shouldn't `code.clone().into_code()` -- this first clones the whole `ContractCode`, and then extracts the vector. It's more minimal to clone just the vector with `codo.code().to_vec()`. Although perhaps the function can be changed to take `code` by value? (if that's a big change, better to keep it out of this PR)

Reply:
got it, fixed